deployment/roles/haystack/files/custom_component.py

```diff
@@ -1,8 +1,36 @@
-from typing import Dict, List
+
+from typing import Dict, List, Optional
+import logging
+
+from haystack.retriever import ElasticsearchRetriever
 from haystack.schema import BaseComponent
 from haystack.retriever.dense import EmbeddingRetriever
 import numpy as np
 from haystack import Document
+logger = logging.getLogger(__name__)
+
+class LabelElasticsearchRetriever(ElasticsearchRetriever):
+
+    def retrieve(self, query: str, filters: dict = None, top_k: Optional[int] = None, index: str = None) -> List[Document]:
+        #  TODO Query labels
+        documents = self.document_store.query(query, None, 1, None, "label_elasticsearch")
+        first_doc = documents[0].to_dict()
+        if len(documents) > 0:
+            logger.warning("retrieving documents from label")
+            documents = self.document_store.get_documents_by_id([first_doc["meta"]["document_id"]], "document_elasticsearch")
```

Comment:
I am not quite sure of what is done here. Are you only taking the first document that is retrieved ? Why not let the possibility to retrieve `k_retriever` documents as it is normally done ?

Reply:
Becasue in this part, we only fetch the document which have the question from the `autocomplete` in its `qas`. We guessed there could be only one doc matching : do you have a different opinion on this ?

Reply:
Yes, in my view, this backend component should be used to retrieve a list of similar questions from the already asked questions. It might be that we have a perfect match, and this is the case that you are trying to tackle here : you suggest a question to the user in the front end so that the question asked matches perfectly what you have in the database. However, I think that this component should also be used to retrieve a `List` of documents that bear questions that are similar to the question the user asked. This would often be the case when the user does not select one of the suggested questions that have already been asked.

Reply:
gotcha.
- longer to dev though : as a first step, I suggest we create another Issue to focus on this.
- we would need to care not to add too many docs for the Reader in terms of perfs